src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -229,6 +229,32 @@ namespace xrt::core::hip
     s_hdl->enqueue(command_cache.get(cmd_hdl));
   }
 
+  // fill data to dst async.
+  template<typename element_type> static void
+  hip_memset_async(void* dst, int value, size_t size, hipStream_t stream)
+  {
+    auto hip_mem_info = memory_database::instance().get_hip_mem_from_addr(dst);
+    auto hip_mem_dst = hip_mem_info.first;
+    auto offset = hip_mem_info.second;
+    assert(hip_mem_dst->get_type() != xrt::core::hip::memory_type::invalid);
+    throw_invalid_value_if(offset + size > hip_mem_dst->get_size(), "dst out of bound in hip_memset!");
```

Comment:
Should we print `dst` pointer also in the error message?

Reply:
fixed.